### ISSUE TITLE
Flexible order and lag random walks

### ISSUE THREAD

**Issue by seabbs** [epiforecasts/epinowcast#135]
It is desirable that random walks are supported with multiple lags and orders. This issue is a placeholder for a full design sketch. See `rw` and `construct_rw` for the relevant functions. 

We can add lags by extending how `enw_cumulative_membership` works (potentially requires a rename). We should also be able to add different orders of random walk by changing the constructed variables.

**Comment by pearsonca** [comment]
note re this: currently the public facing function (`rw(...)`) documentation refers to the internal function (`construct_rw(...)`) arguments.

Probably makes sense to split into public vs internal elements, but will have to figure out someway to preserve that split while exposing the right amount of internal interface.

**Comment by seabbs** [comment]
Does it? I don't think it does?

**Comment by seabbs** [comment]
I don't think it is a problem directing users to internal functions so they can understand them/how the package works but without making it external (as clearly not for use).

**Comment by seabbs** [comment]
I think we can do a second order random walk (i.e on the differences by making a cumulative sum of how many timesteps have passed within the random walk period. I.e. for a weekly random walk within a week we would have 1, 2, 3,4,5,6,7 rather than 1,1,1,1,1,1,1 as we do now. This would mean the end of the week has 7 times the same step size and you end up getting a piecewise linear residual with a random walk on the slope.
